python/smashbox/reporter/__init__.py

```diff
@@ -0,0 +1,132 @@
+import sys
+
+class Reporter:
+    """ Report execution state of smashbox.
+    """
+
+    def __init__(self):
+        self.LOG=True
+
+    def smashbox_start(self,args,config):
+        """
+        Smashbox is starting.
+        Arguments:
+         - args: Namespace object with all the options passed when invoking smash executable
+         - config: global configuration object 
+        """
+
+        if self.LOG:
+            print "SMASHBOX_START",args,config
+            self.config=config
+        curl_check_url(config)
+        check_owncloudcmd(config)
+    def smashbox_stop(self):
+        """
+        Smashbox is about to stop.
+        """
+
+        if self.LOG:
+            print "SMASHBOX_STOP"
+
+    
+    def testcase_start(self,name,loop_i,testset_i,namespace):
+        """
+        Testcase is about to start.
+        Arguments:
+         - name: short name of the testcase 
+         - loop_i: loop index or None if not running in the loop
+         - testset_i: testset index or None if running with default configuration
+         - namespace: access to the testcase module namespace
+
+         Example: 
+          try:
+           print "Current testset configuration", namespace.testsets[testset_i]
+          expect AttributeError:
+           print "Testsets not defined"
+
+        """
+
+        if self.LOG:
+            print "TESTCASE_START",name,loop_i,testset_i,namespace.__doc__
+
+            barename=name.replace("test_","")
+
+            for c in self.config.__dict__:
+                if c.startswith(barename+"_"):
+                    print c,self.config[c]
+
+
+    def testcase_stop(self,returncode):
+        if self.LOG:
+            print "TESTCASE_STOP",returncode
+
+def check_owncloudcmd(config):
+    import subprocess
+    from smashbox.utilities import  oc_webdav_url
+    #create tmp directory localy
+    cmd = 'mkdir -p test'
+    process = subprocess.Popen(cmd, shell=True,stdout=subprocess.PIPE,stderr=subprocess.PIPE)
+    process.wait()
+    #check if owncloudcmd is correct or if path to owncloudcmd is not correct
+    cmd = config.oc_sync_cmd+' '+'tests'+' '+oc_webdav_url('owncloud',remote_folder='test', user_num=None)
+    process = subprocess.Popen(cmd, shell=True,stdout=subprocess.PIPE,stderr=subprocess.PIPE)
+    stdout,stderr = process.communicate()
+    if((stderr).find("not found") != -1):
+        print stderr
+        sys.exit()
+    #delete tmp directory localy
+    cmd = 'rm -rf test'
+    process = subprocess.Popen(cmd, shell=True,stdout=subprocess.PIPE,stderr=subprocess.PIPE)
+    process.wait()
+    
+def curl_check_url(config):
```

Comment:
Maybe you should consider using the curl client built in the smashbox? I also intend to migrate to this client instead of the CLI curl command https://github.com/cernbox/smashbox/blob/master/python/smashbox/curl.py

Here is an example how the client is already used in other parts of smashbox:
https://github.com/cernbox/smashbox/blob/master/protocol/protocol.py#L182